### PR TITLE
Refactor PostcodeConditionTests

### LINKED ISSUE #24: Refactor Postcode Tests
### Description

We have recently added more countries in our postcode validation. We should refactor our tests to make it more DRY and robust as we continue to add countries. See [the discussion on PR #17](https://github.com/ustwo/formvalidator-swift/pull/17#discussion_r89733545).

### PR DESCRIPTION
**Pull request by ghost**
Resolves #24 `Refactor Postcode Tests`.

### Summary

Refactored `PostcodeConditionTests`.

### Details

- Create extendible `PostcodeConditionTests` for when we add new countries
- Add tests for United States post codes
- Add test for changing countries in `PostcodeCondition`
- Update `PostcodeCountries` to include `allValues` convenience var
- Make `PostcodeCountries.regex` public
- Alphabetise enumeration values.

### CODE REVIEW

File: Tests/Unit Tests/Conditions/PostcodeConditionTests.swift [hunk 1]
```diff
@@ -14,189 +14,96 @@ import XCTest
...
+                        "12345-1",
+                        nil]
+            }
+        }
+        
+        init(country: PostcodeCountries) {
+            self.country = country
+            condition = PostcodeCondition(country: country)
+        }
+
```
Comment (madhikarma) [inline Tests/Unit Tests/Conditions/PostcodeConditionTests.swift:71]: Remove newline?

**Review by madhikarma — APPROVED** [review]
Looks cool @aamctustwo Saw one style misdemeanour but that's it. An interesting approach to reusable tests

File: Tests/Unit Tests/Conditions/PostcodeConditionTests.swift (hunk 1 above)
Comment (ghost) [reply Tests/Unit Tests/Conditions/PostcodeConditionTests.swift:71]: Convention in this project has been to have an empty line between the end of the last function/property/etc. and the ending closing brace of the class/struct/enum/etc.

**Comment by ghost** [comment]
@madhikarma The updated tests are based (very loosely) to things I've seen in the way SwiftLint handles its tests.

**Pull request merged**